clients/unshaded/src/main/java/tachyon/client/file/URIStatus.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.util.List;
+
+import tachyon.annotation.PublicApi;
+import tachyon.thrift.FileInfo;
+
+/**
+ * Wrapper around {@link FileInfo}. Represents the metadata about a file or directory in Tachyon.
+ * This is a snapshot of information about the file or directory and not all fields are guaranteed
+ * to be up to date. Fields documented as immutable will always be accurate, and fields
```

Comment:
Annotations would be a better way of keeping track, not sure if there is something nice already we can use (since mutable/immutable should be pretty common). I've updated the naming